src/main/java/frc/robot/subsystems/Lift.java

```diff
@@ -307,8 +313,35 @@ public void initialize() {
             - Cal.Lift.ELEVATOR_ABS_ENCODER_POS_AT_START_INCHES);
   }
 
+  /**
+   * if the robot has completed startScore but hasn't started finishScore, then stop the robot from
+   * scoring
+   */
   public void cancelScore() {
-    // TODO do this
+    // means startScore has run
+    if (desiredPosition == LiftPosition.PRE_SCORE_HIGH_CONE
+        || desiredPosition == LiftPosition.PRE_SCORE_MID_CONE) {
+      // but finishScore hasn't run
+      if (desiredGrabberClosed) {
+        // check to see if startScore had changed the desiredPosition to HIGH or MID
+        if (desiredPosition == LiftPosition.PRE_SCORE_HIGH_CONE) {
+          setDesiredPosition(LiftPosition.SCORE_HIGH_CONE);
+        } else {
+          setDesiredPosition(LiftPosition.SCORE_MID_CONE);
```

Comment:
Why do we do this?

Reply:
I could be wrong, but I added this because I thought that before startScore() was called, the liftPosition could have been either high or mid. I'm not completely sure though.

Reply:
Well it can, you're right about that, but do we want the desired position to actually be changed to the scoring position? I'm genuinely asking, I don't know the answer. So I can't say what's right or wrong. But just make sure you think about whether that's necessary.

Reply:
I think Basel told me that cancelScore should essentially undo what startScore did, so then I think that would be right. I could be misremembering though.

Reply:
Ok, sounds good. Just that one minor change to `this.cancel()`, then.